lib/virt_disk/block_file.rb

```diff
@@ -1,9 +1,10 @@
 module VirtDisk
   class BlockFile
-    attr_accessor :path
+    attr_accessor :path, :offset
 
-    def initialize(path)
+    def initialize(path, offset=0)
```

Comment:
@roliveri this is mostly a convenience mechanism, since this block code may be invoked in any number of places, adding this offset to each call at that level would be tedious & result in alot of duplicate code.

This way the higher level code can set this offset once if desired (implementing a barebones MBR/GPT parser for example and setting the offset from a partition address) and not worry about it.